### ISSUE TITLE
SWIG + Scilab 6.0.0-beta2 example from documentation fails

### ISSUE THREAD

**Issue by accesio** [swig/swig#746]
I downloaded the scilab-6.0.0-beta-2

I am using swig 3.0.10 release

Using the tutorial, 

%module example

%inline %{
double Foo = 3.0;

int fact(int n) {
    if (n < 0) {
        return 0;
    }
    else if (n == 0) {
        return 1;
    }
    else {
        return n * fact(n-1);
    }
}
%}

example.i 

```bash
gcc -fPIC -c -I/usr/local/include/scilab example_wrap.c
gcc -shared example_wrap.o -o libexample.so
```

I then invoke

exec loader.sce 

which appears to load the libexample.so into scilab

Then, I type

fact(5) 

and scilab segmentation faults:

A fatal error has been detected by Scilab.
Please check your user-defined functions (or external module ones) should they appear in the stack trace.
Otherwise you can report a bug on http://bugzilla.scilab.org/ with:
 * a sample code which reproduces the issue
 * the result of [a, b] = getdebuginfo()
 * the following information:
[paris:28260] Signal: Segmentation fault (11)
[paris:28260] Signal code: Address not mapped (1)
[paris:28260] Failing at address: 0x1

Call stack:
   1: 0x914da6 <JVM_handle_linux_signal>        (/home/jdamon/Tools/Scilab/scilab-6.0.0-beta/thirdparty/java//lib/amd64/server/libjvm.so)
   2: 0x90b6f3 < >                              (/home/jdamon/Tools/Scilab/scilab-6.0.0-beta/thirdparty/java//lib/amd64/server/libjvm.so)
   3: 0x10330  < >                              (/lib/x86_64-linux-gnu/libpthread.so.0)
   4: 0x1e07db <getNbInputArgument>             (/home/jdamon/Tools/Scilab/scilab-6.0.0-beta/lib/scilab/libscilab-cli.so.6)
   5: 0x1e08f1 <checkInputArgument>             (/home/jdamon/Tools/Scilab/scilab-6.0.0-beta/lib/scilab/libscilab-cli.so.6)
   6: 0x3b32   <_wrap_fact>                     (/media/jdamon/Development/Documents/Projects/aiousb_patrick_mcbride_issue/AIOUSB/AIOUSB/lib/wrappers/scilab/libexample.so)
   7: 0x3c2717 <types::WrapCFunction::call(std::vector<types::InternalType*, std::allocator<types::InternalType*> >&, std::unordered_map<std::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t> >, types::InternalType*, std::hash<std::basic_string<wch> (/home/jdamon/Tools/Scilab/scilab-6.0.0-beta/lib/scilab/libsciast.so.6)
   8: 0x3b7908 <types::Callable::invoke(std::vector<types::InternalType*, std::allocator<types::InternalType*> >&, std::unordered_map<std::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t> >, types::InternalType*, std::hash<std::basic_string<wchar_> (/home/jdamon/Tools/Scilab/scilab-6.0.0-beta/lib/scilab/libsciast.so.6)
   9: 0x1c19d4 <ast::RunVisitorT<ast::ExecVisitor>::visitprivate(ast::CallExp const&)> (/home/jdamon/Tools/Scilab/scilab-6.0.0-beta/lib/scilab/libsciast.so.6)
  10: 0x1bbcec <ast::RunVisitorT<ast::ExecVisitor>::visitprivate(ast::SeqExp const&)> (/home/jdamon/Tools/Scilab/scilab-6.0.0-beta/lib/scilab/libsciast.so.6)
  11: 0x1d9309 <StaticRunner::launch()>         (/home/jdamon/Tools/Scilab/scilab-6.0.0-beta/lib/scilab/libscilab-cli.so.6)
  12: 0x1d1792 <RunScilabEngine>                (/home/jdamon/Tools/Scilab/scilab-6.0.0-beta/lib/scilab/libscilab-cli.so.6)
  13: 0x20c9   <main>                           (scilab-bin)
  14: 0x21f45  <__libc_start_main>              (/lib/x86_64-linux-gnu/libc.so.6)
  15: 0x2631   < >                              (scilab-bin)
End of stack

**Comment by ojwb** [comment]
Fix committed, so closing.

**Comment by smarchetto** [comment]
fixed by [d0a45be](https://github.com/swig/swig/commit/d0a45be1ebfe125d51696b48cb5d359bc7431052)
